src/main/java/baritone/behavior/LookBehavior.java

```diff
@@ -126,8 +128,12 @@ public void pig() {
         }
     }
 
-    public Rotation getEffectiveRotation() {
-        return this.serverRotation;
+    public Optional<Rotation> getEffectiveRotation() {
+        if (Baritone.settings().freeLook.value || Baritone.settings().blockFreeLook.value) {
+            return Optional.of(this.serverRotation);
```

Comment:
Doesn't `Optional.of` throw an NPE if you pass it `null` and this should be `Optional.ofNullable` to turn the `null` into an `Optional.empty()`?

Reply:
fixed lol